packages/react-native/Libraries/Blob/URLSearchParams.js

```diff
@@ -11,60 +11,129 @@
 // Small subset from whatwg-url: https://github.com/jsdom/whatwg-url/tree/master/src
 // The reference code bloat comes from Unicode issues with URLs, so those won't work here.
 export class URLSearchParams {
-  _searchParams: Array<[string, string]> = [];
+  _searchParams: Map<string, string[]> = new Map();
 
-  constructor(params?: Record<string, string>) {
-    if (typeof params === 'object') {
-      Object.keys(params).forEach(key => this.append(key, params[key]));
+  constructor(params?: Record<string, string> | string | [string, string][]) {
```

Comment:
Let's validate `params` before proceeding (typeof null is 'object'!)

```suggestion
  constructor(params?: Record<string, string> | string | [string, string][]) {
    if (params == null) {
      return;
    }
```

Reply:
Added null check .